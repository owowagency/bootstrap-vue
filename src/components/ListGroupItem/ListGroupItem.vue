<template>
    <Component
        :is="tag"
        class="list-group-item"
        :class="classes"
    >
        <!-- @slot Displays the content of the list group item -->
        <slot />
    </Component>
</template>

<script lang="ts">
import useActive, {activeProps} from '@/composables/useActive';
import useDisabled, {disabledProps} from '@/composables/useDisabled';
import useVariant, {variantProp} from '@/composables/useVariant';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...activeProps,
    ...disabledProps,
    /**
     * The tag to be used for the list group item
     */
    tag: {
        type: String,
        default: 'li',
    },
    variant: variantProp(''),
});


const {classes} = useClasses(computed(() => [
    ['a', 'button'].includes(props.tag) ? 'list-group-item-action' : undefined,
    useActive(props.active).activeClass.value,
    useDisabled(props.disabled).disabledClass.value,
    useVariant(props.variant, 'list-group-item-{0}').variantClass.value,
]));
</script>

<docs>
```vue
<ListGroupItem>
    List group item
</ListGroupItem>
```

## Imported Props

| Prop name | Description                  | Type                                             | Values | Default |
| --------- | ---------------------------- | ------------------------------------------------ | ------ | ------- |
| active    | Adds the `active` class      | boolean                                          | -      | false   |
| disabled  | Disables the component       | boolean                                          | -      | false   |
| variant   | The variant of the component | [Variant](../../composables/useVariant) (string) | -      | ''      |
</docs>
