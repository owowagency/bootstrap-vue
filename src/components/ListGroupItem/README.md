# ListGroupItem

```vue
<ListGroupItem>
    List group item
</ListGroupItem>
```

## Imported Props

| Prop name | Description                  | Type                                             | Values | Default |
| --------- | ---------------------------- | ------------------------------------------------ | ------ | ------- |
| active    | Adds the `active` class      | boolean                                          | -      | false   |
| disabled  | Disables the component       | boolean                                          | -      | false   |
| variant   | The variant of the component | [Variant](../../composables/useVariant) (string) | -      | ''      |

## Props

| Prop name | Description                                | Type            | Values | Default |
| --------- | ------------------------------------------ | --------------- | ------ | ------- |
| tag       | The tag to be used for the list group item | string          | -      | 'li'    |
| variant   |                                            | variantProp('') | -      |         |

## Slots

| Name    | Description                                 | Bindings |
| ------- | ------------------------------------------- | -------- |
| default | Displays the content of the list group item |          |
