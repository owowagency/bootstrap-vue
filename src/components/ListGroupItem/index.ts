import ListGroupItem from './ListGroupItem.vue';

export default ListGroupItem;
