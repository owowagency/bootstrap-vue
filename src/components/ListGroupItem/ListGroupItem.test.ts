import ListGroupItem from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(ListGroupItem);

    componentSlotRenderTest(ListGroupItem);

    componentWrapperClassTest(
        ListGroupItem,
        {tag: 'a'},
        'list-group-item-action',
        'adds list-group-item-class when tag is a',
    );

    componentWrapperClassTest(
        ListGroupItem,
        {tag: 'button'},
        'list-group-item-action',
        'adds list-group-item-class when tag is button',
    );

    componentWrapperClassTest(ListGroupItem, {active: true}, 'active');

    componentWrapperClassTest(ListGroupItem, {variant: 'danger'}, 'list-group-item-danger');

    componentWrapperTagTest(ListGroupItem, {props: {tag: 'table'}});
});
