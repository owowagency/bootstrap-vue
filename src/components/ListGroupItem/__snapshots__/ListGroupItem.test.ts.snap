// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<li class=\\"list-group-item list-group-item-\\">
  <!-- @slot Displays the content of the list group item -->
</li>"
`;
